apiserver/params/crossmodel.go

```diff
@@ -22,3 +22,47 @@ type CrossModelOffer struct {
 type CrossModelOffers struct {
 	Offers []CrossModelOffer `json:"offers"`
 }
+
+// RemoteEndpoint holds information about remote endpoint.
+type RemoteEndpoint struct {
+	// Name is exported relation name.
+	Name string `json:"name"`
+
+	// Interface is exported relation interface.
+	Interface string `json:"interface"`
+
+	// Role is exported relation role.
+	Role string `json:"role"`
+}
+
+// RemoteServiceInfoResult holds remote service information for a URL or
+// a related error.
+type RemoteServiceInfoResult struct {
+	// URL contains remote service URL.
+	URL string `json:"url"`
+
+	// RemoteService contains remote service information.
+	RemoteService RemoteServiceInfo `json:"remote_service"`
```

Comment:
I made Error and Results as a pointers. I have not sen anyway where the Result is a pointer (in this older diff it's still called RemoteService, later I've changed to Result for consistency)